### PR TITLE
[REVIEW] Apply `na_rep` to column names in csv writer

### PR DESCRIPTION
**Pull request by galipremsagar**
Fixes: #6688 

`na_rep` was not being applied to column names, column names can be `None` too. So this PR introduces changes that will apply `na_rep` for column names as well.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by galipremsagar** [comment]
rerun tests

File: python/cudf/cudf/_lib/csv.pyx [hunk 1]
```diff
@@ -427,7 +427,11 @@ cpdef write_csv(
     cdef sink_info sink_info_c = make_sink_info(path_or_buf, data_sink_c)
 
     if header is True:
-        all_names = table._column_names
+        all_names = tuple(
+            table.columns.to_native_types(
+                na_rep=na_rep,
+            )
+        )
```
Comment (kkraus14) [inline python/cudf/cudf/_lib/csv.pyx:434]: Should we implement this without `.columns`? That constructs a Pandas Index object which we've seen to be kinda expensive.

Comment (galipremsagar) [reply python/cudf/cudf/_lib/csv.pyx]: Changed the implementation to use `_column_names` instead of creating Index.

**Review by kkraus14 — APPROVED** [review]

**Pull request merged**